solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -687,36 +693,42 @@ contract RandomBeacon is Ownable {
         }
     }
 
-    /// @notice Creates a new relay entry.
+    /// @notice Creates a new relay entry. Parameters `groupPubKey` and
+    ///         `groupMembers` must match to the fields of the stored group
+    ///         chosen to submit the relay entry.
     /// @param submitterIndex Index of the entry submitter.
     /// @param entry Group BLS signature over the previous entry.
-    function submitRelayEntry(uint256 submitterIndex, bytes calldata entry)
-        external
-    {
+    /// @param groupPubKey Public key of the group chosen to submit the
+    ///        relay entry.
+    /// @param groupMembers Identifiers of members of the group chosen to
+    ///        submit the relay entry.
+    function submitRelayEntry(
+        uint256 submitterIndex,
+        bytes calldata entry,
+        bytes calldata groupPubKey,
+        uint32[] calldata groupMembers
+    ) external {
         uint256 currentRequestId = relay.currentRequest.id;
 
-        Groups.Group memory group = groups.getGroup(
-            relay.currentRequest.groupId
-        );
-
         uint256 slashingAmount = relay.submitEntry(
             sortitionPool,
             submitterIndex,
             entry,
-            group
+            groupPubKey,
+            groupMembers,
+            groups.getGroupChecksum(relay.currentRequest.groupId)
```

Comment:
Yep, it's cheaper. This change lowers the cost of `submitRelayEntry` from 245k to 215k. The most expensive part while getting a `Group` struct is reading `members` field from storage and allocating memory for it. Getting a group without that field is quite cheap.

Reply:
cc @nkuba

Reply:
We could try pulling the members as a storage reference.
```solidity
    function submitRelayEntry(
        uint256 submitterIndex,
        bytes calldata entry,
        bytes calldata groupPubKey
    ) external {
        uint256 currentRequestId = relay.currentRequest.id;

        uint32[] storage groupMembers = groups.getGroup(groupPubKey).members;

        uint256 slashingAmount = relay.submitEntry(
            sortitionPool,
            submitterIndex,
            entry,
            groupPubKey,
            groupMembers
        );
 (...)
```
This will require `getGroup` functions returns to be replaced from `Group memory` to `Group storage`.

With these changes, I was able to reduce `submitRelayEntry` gas cost from `216566` to `199064` in one of the simplest test cases. Please note that we may notice some contract size problems.

Reply:
Got rid of group checksums and applied the proposed approach in https://github.com/keep-network/keep-core/pull/2736/commits/ff4a38f2fc998748bd48dabf2693f4ef3e922322. Gas usage of `submitRelayEntry` is now:
```
·-------------------------------------------------------------|---------------------------|-------------|-----------------------------·
|                     Solc version: 0.8.6                     ·  Optimizer enabled: true  ·  Runs: 200  ·  Block limit: 30000000 gas  │
······························································|···························|·············|······························
|  Methods                                                    ·               95 gwei/gas               ·       4153.04 eur/eth       │
·························|····································|·············|·············|·············|···············|··············
|  Contract              ·  Method                            ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
·························|····································|·············|·············|·············|···············|··············
|  RandomBeaconStub      ·  submitRelayEntry                  ·     207339  ·     494437  ·     301313  ·           22  ·     118.88  │
·························|····································|·············|·············|·············|···············|··············
```